framework/src/packets/ip/v6/srh.rs

```diff
@@ -134,7 +134,21 @@ pub struct BadSegmentsError;
     display = "Segments left {} must be less than segment list length {}",
     _0, _1
 )]
-pub struct SegmentsLeftOutOfBoundError(u8, usize);
+pub struct SegmentsLeftOutOfBoundsError(u8, usize);
+
+/// Error for bad index within segment list
+#[derive(Debug, Fail)]
+#[fail(display = "Index {} out of bounds of segment list length {}", _0, _1)]
+pub struct SegmentsIndexOutOfBoundsError(usize, usize);
+
+/// Trait with helper functions for segment list traversing/choosing.
+pub trait SegmentsIndex {
```

Comment:
just wondering, should we make this the generic extension trait for SegmentRouting? we don't have anything else to dump in here now but it gives us a place later if we do and don't need changes to calling code down the road.

https://github.com/nox/rust-rfcs/blob/master/text/0445-extension-trait-conventions.md

Reply:
Sure.